js/msp.js

```diff
@@ -297,10 +297,12 @@ var MSP = {
                 CONFIG.mode = data.getUint32(6, 1);
                 CONFIG.profile = data.getUint8(10);
                 CONFIG.cpuload = data.getUint16(11, 1);
-                CONFIG.numProfiles = data.getUint8(13);
-                CONFIG.rateProfile = data.getUint8(14);
+                if (semver.gt(CONFIG.flightControllerVersion, "2.9.1")) {
+                    CONFIG.numProfiles = data.getUint8(13);
+                    CONFIG.rateProfile = data.getUint8(14);
 
-                TABS.pid_tuning.checkUpdateProfile(true);
+                    TABS.pid_tuning.checkUpdateProfile(true);
```

Comment:
This shouldn't be conditional. Profile switching is supported by earlier versions as well, and there are guards for the rateprofile related functionality inside the method.